### PR TITLE
fix sending binaryData media (photo, document, video etc.) issues on Telegram Node - N8N-3783

### PR DESCRIPTION
**Pull request by tahasonmez**
- added necessary headers while sending multipart/form-data requests
- Since Telegram refuse the request if file name is undefined/null/empty, 'file name' is added as option when binaryData is selected

https://community.n8n.io/t/telegram-wrong-url-host-while-sending-binary-data/14470

### CODE REVIEW

File: packages/nodes-base/nodes/Telegram/Telegram.node.ts [hunk 1]
```diff
@@ -801,7 +802,31 @@ export class Telegram implements INodeType {
 				placeholder: '',
 				description: 'Name of the binary property that contains the data to upload',
 			},
-
+			{
+				displayName: 'File Name',
+				name: 'fileName',
+				type: 'string',
+				default: '',
+				required: true,
```
[the remark this answers is not in the document]
Comment (tahasonmez) [reply packages/nodes-base/nodes/Telegram/Telegram.node.ts]: good point, file name is optional now.

**Comment by michael-radency** [comment]
Hello @tahasonmez thank you for your contribution, I did add some small improvements

File: packages/nodes-base/nodes/Telegram/Telegram.node.ts [hunk 2]
```diff
@@ -2187,19 +2176,27 @@ export class Telegram implements INodeType {
...
 							},
 						},
 					};
-					responseData = await apiRequest.call(this, requestMethod, endpoint, {}, qs, { formData });
+
+					const formDataOptions = new FormData(formData);
+
+					const options = {headers: formDataOptions.getHeaders(), formData};
+
+					responseData = await apiRequest.call(this, requestMethod, endpoint, {}, qs, options);
```
Comment (michael-radency) [inline packages/nodes-base/nodes/Telegram/Telegram.node.ts:2199]: @tahasonmez 
Can you please explain those changes to how formData is passed? Isn't It working as it was before?

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with `n8n@0.186.0`

**Pull request merged**